### PR TITLE
Delete Flavor API

### PR DESCRIPTION
**Pull request by lvillasica**
JIRA: https://musashi.atlassian.net/browse/AVIATORAPI-91

### CODE REVIEW

**Comment by jocaba** [comment]
+1 Looks okay to me. I'd just like to note that Openstack's API docs are wrong about this one. Martin's POST request was the needed magic ingredient

File: test/aviator/openstack/compute/v2/admin/delete_flavor_test.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      request = klass.new(session_data) do |p|
+        p[:flavor_id] = flavor_id
+      end
+
+      request.url.must_equal url
+    end
+
+
+    validate_response 'valid params are provided' do
+      flavor    = session.compute_service.request(:list_flavors).body[:flavors].last
```
Comment (stefanymarie) [inline test/aviator/openstack/compute/v2/admin/delete_flavor_test.rb]: suggestion: create the flavor to be deleted

Comment (choodur) [inline test/aviator/openstack/compute/v2/admin/delete_flavor_test.rb]: +1. this would make the test predictable and consistent.

Comment (lvillasica) [inline test/aviator/openstack/compute/v2/admin/delete_flavor_test.rb]: @choodur @stefanymarie will need the create flavor api to be merged for that.

Comment (stefanymarie) [inline test/aviator/openstack/compute/v2/admin/delete_flavor_test.rb]: @lvillasica, already merged https://github.com/musashi-dev/aviator/pull/52

**Comment by lvillasica** [comment]
@musashi-dev/ronins please check again. :smile_cat:

**Comment by choodur** [comment]
+1, this is okay now.

**Pull request merged**